packages/flutter_tools/lib/src/artifacts.dart

```diff
@@ -37,17 +44,36 @@ String _artifactToFileName(Artifact artifact) {
       return 'vm_isolate_snapshot.bin';
     case Artifact.isolateSnapshotData:
       return 'isolate_snapshot.bin';
+    case Artifact.platformKernelDill:
+      return 'platform.dill';
+    case Artifact.platformLibrariesJson:
+      return 'libraries.json';
+    case Artifact.flutterPatchedSdkPath:
+      assert(false, 'No filename for sdk path, should not be invoked');
+      return null;
+    case Artifact.engineDartSdkPath:
+      return 'dart-sdk';
+    case Artifact.frontendServerSnapshotForEngineDartSdk:
+      return 'frontend_server.dart.snapshot';
   }
   assert(false, 'Invalid artifact $artifact.');
   return null;
 }
 
+class EngineBuildPaths {
+  String targetEngine;
+  String hostEngine;
+  EngineBuildPaths({@required this.targetEngine, @required this.hostEngine}) {
```

Comment:
Introduced assert initializers, which required adding '--assert-initializer' to the list of VM options for flutter invocation script.